tests/unit/test_api_helpers.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Tests for our API helper methods."""
+
+import unittest
+
+import webcompat
+from webcompat.api.helpers import get_html_comments
+
+
+class TestAPIHelpers(unittest.TestCase):
+    """Class for testing API helpers."""
+
+    def setUp(self):
+        """Set up the tests."""
+        webcompat.app.config['TESTING'] = True
+        self.app = webcompat.app.test_client()
+
+    def tearDown(self):
+        """Tear down the tests."""
+        pass
+
+    def test_get_html_comments_single_comment(self):
+        """Test we get a template returned for a single comment."""
+        response = (b'{"user": {"login": "miketaylr","avatar_url": "https://avatars1.githubusercontent.com/u/67283?v=4"},"created_at": "2016-07-28T14:41:43Z","body_html": "<p>test</p>"}',  # noqa
+            200, {'content-type': 'application/json'})
+        with webcompat.app.app_context():
+            rv = get_html_comments(response)
+            assert b'<p>test</p>' in rv.data
+            assert rv.status_code == 200
+            assert rv.content_type == 'text/html'
+
+    def test_get_html_comments_multi_comment(self):
+        """Test we get a template returned for multiple comments."""
+        response = (b"""[
+            {"user": {"login": "miketaylr","avatar_url": "https://avatars1.githubusercontent.com/u/67283?v=4"},"created_at": "2016-07-28T14:41:43Z","body_html": "<p>one</p>"},
+            {"user": {"login": "miketaylr","avatar_url": "https://avatars1.githubusercontent.com/u/67283?v=4"},"created_at": "2016-07-28T14:42:43Z","body_html": "<p>two</p>"}]""",  # noqa
+            200, {'content-type': 'application/json'})
+        with webcompat.app.app_context():
+            rv = get_html_comments(response)
+            assert b'<p>one</p>' in rv.data
+            assert b'<p>two</p>' in rv.data
+            assert rv.status_code == 200
+            assert rv.content_type == 'text/html'
+
```

Comment:
304 actually fails... 😈 

I think it's because in the normal case, GitHub sends back an empty response body (`b''`) but the browser pulls the data from its HTTP cache so the templates don't throw. Maybe the right thing to do is just read from a fixture file to mimic a read from the HTTP cache. I'll do that.

Reply:
OK, I re-wrote how we handle this, with a `try... except`. If we fail to parse due to 304, we should still survive.